src/main/java/org/broadinstitute/hellbender/tools/exome/PoNIssueDetector.java

```diff
@@ -0,0 +1,163 @@
+            final Map<String, Double> contigToMedian = getContigToMedianCRMap(qcTangentNormalization.getTangentNormalized());
+
+            // Segment the result
+            // For each tangent normalization result from the QC PoN
+            if (ctx == null) {
+                final List<ReadCountCollection> singleSampleTangentNormalizedReadCounts = createIndividualReadCountCollections(qcTangentNormalization.getTangentNormalized());
+                result = identifySamplesWithSuspiciousContigs(singleSampleTangentNormalizedReadCounts, contigToMedian);
+            } else {
+                final JavaRDD<ReadCountCollection> parallelSingleSampleTangentNormalizedReadCounts = createParallelIndividualReadCountCollections(qcTangentNormalization.getTangentNormalized(), ctx);
+                result = identifySamplesWithSuspiciousContigs(parallelSingleSampleTangentNormalizedReadCounts, ctx, contigToMedian);
+            }
+        }
+
+        return result;
+    }
+
+    @VisibleForTesting
+    static Map<String, Double> getContigToMedianCRMap(final ReadCountCollection readCountCollection) {
+        final List<String> allContigsPresent = retrieveAllContigsPresent(readCountCollection);
+        final Map<String, Double> contigToMedian = new LinkedHashMap<>();
+        for (String contig: allContigsPresent) {
+            final ReadCountCollection oneContigReadCountCollection = readCountCollection.subsetTargets(readCountCollection.targets().stream().filter(t -> t.getContig().equals(contig)).collect(Collectors.toSet()));
+            final double[] flatCounts = Doubles.concat(oneContigReadCountCollection.counts().getData());
+
+            // Put into CRSpace
+            final double[] flatCountsInCRSpace = DoubleStream.of(flatCounts).map(d -> Math.pow(2, d)).toArray();
+            contigToMedian.put(contig, new Median().evaluate(flatCountsInCRSpace));
+        }
+        return contigToMedian;
+    }
+
+    /**
+     *  Given a single sample tangent normalization (or other coverage profile), determine whether any contigs look like
+     *   it has an arm level event (defined as 25% (or more) of the contig amplified/deleted)
+     *
+     * @param singleSampleTangentNormalized Tangent normalized data for a single sample.
+     * @return never {@code null}
+     */
+    private static Boolean hasSuspiciousContigs(final ReadCountCollection singleSampleTangentNormalized, final Map<String, Double> contigToMedian) {
+
+        final List<String> allContigsPresent = retrieveAllContigsPresent(singleSampleTangentNormalized);
+        for (String contig: allContigsPresent) {
+            final ReadCountCollection oneContigReadCountCollection = singleSampleTangentNormalized.subsetTargets(singleSampleTangentNormalized.targets().stream().filter(t -> t.getContig().equals(contig)).collect(Collectors.toSet()));
+            final RealMatrix counts = oneContigReadCountCollection.counts();
```

Comment:
Might as well work with a `RealVector` instead of a matrix starting from this line.